source/computer/wsl-setup.rst

```diff
@@ -1,18 +1,154 @@
 WSL 配置指南
 ============
 
+:本节贡献者: |赵志远|\（作者）、
+             |田冬冬|\（审稿）、
+             |姚家园|\（审稿）
+:最近更新日期: 2021-01-18
+:预计花费时间: 120 分钟
+
+WSL 是什么?
+-------------
+
+Windows Subsystem for Linux（简称 WSL）是一个在 Windows 10 上运行原生
+Linux 二进制可执行文件（ ELF 格式）的兼容层，也被称为“适用于 Linux 的 Windows 子系统”。
+
+启用 WSL 功能后，就可以在 Windows 上安装 Linux 发行版，原生运行 Linux 下的大多数
+命令行工具、实用工具和应用程序。WSL 可以同时安装多个不同的 Linux 发行版，并且随时
+进行切换，支持利用 ``wsl --export`` 命令将整个子系统随时打包备份，并在之后
+利用 ``wsl --import`` 命令进行还原。
+
+与安装双系统相比，WSL 安装简便且切换方便。与使用虚拟机相比，WSL 在启动速度、
+消耗资源以及性能利用上有较大优势。因此，对于不想安装双系统或者虚拟机的用户来说，
+WSL 是个不错的选择。
+
+WSL 有 WSL1 和 WSL2 两个发行版本，二者底层原理不同。大多数情况下，建议使用 WSL2，
+因为它提供更快的性能和 100% 的系统调用兼容性。但是，当项目需要经常从 Linux 上访问存储
+在 Windows 系统上的文件，或利用 Windows 软件对 Linux 上的文件进行处理时，使用 WSL1 具有
+更快实现的性能。
+
+官方目前没有弃用 WSL1 的计划，并且支持将任何一个已经安装的 Linux 发行版更改为 WSL1 或者 WSL2。
+
+安装 WSL
+---------------
+
+WSL 的安装方式可以参考官方文档：
+
+- 中文指南：https://docs.microsoft.com/zh-cn/windows/wsl/install-win10
+- 英文指南：https://docs.microsoft.com/en-us/windows/wsl/install-win10
+
+官方文档对 WSL 安装配置中可能遇到的常见问题都给出了详细的解决方案，请认真阅读。
+
+WSL 可以安装不同的 Linux 发行版，但目前官方并未提供 Fedora 发行版的安装。
+推荐使用 WSL 安装 Ubuntu 20.04 LTS 或 Ubuntu 20.10。
+
 .. note::
 
-   尚未完成。
+   开启 WSL 后 Linux 发行版的默认安装位置在 C 盘。若安装多个发行版，
+   则会占用 C 盘的大量空间。可以先将一个新安装的 Linux 发行版导出备份，
+   接着再还原到其它的位置，然后将 C 盘上的发行版删除。这样可以保证 C 盘
+   有足够的空间。
+
+配置 Linux
+-----------
+
+安装 WSL 之后，还需要对 Linux 发行版进行一定配置。
+
+如果你安装了 Ubuntu，可以参考 《\ :doc:`/computer/ubuntu-setup` 》进行配置。
 
-   WSL 的安装方式可以参考：
+安装 X Server
+--------------
 
-   - 中文指南：https://blog.walterlv.com/post/how-to-install-wsl2.html
-   - 英文指南：https://docs.microsoft.com/en-us/windows/wsl/install-win10
+WLS 本身不支持图形界面，需要在 Windows 中安装 X server
+来接收和显示 Linux 中的图形界面。
 
 .. note::
 
-   由于维护者目前没有可供安装 Windows 的空闲机器，因而本文短期内无法完成。
-   如果你有 Windows 使用经验，且愿意从头安装 WSL，请到
-   `Issue #65 <https://github.com/seismo-learn/seismology101/issues/65>`__
-   下发表评论。
+   X Window System（常称为 X11 或 X）是 UNIX 系统下常用的一种视窗系统，
+   主要由 X Server 和 X Client 两部分组成。其中 X Server 负责接受对图形输出
+   的请求并反馈用户输入，而 X Client 则是使用图形界面的应用程序。
+
+
+Windows 下常见的 X Server 有 `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`__\ 、
+`Xming <http://www.straightrunning.com/XmingNotes/>`__\ 、
+`Xmanager <https://www.xshellcn.com/>`__ 等。
+其中，VcXsrv 是开源免费软件；Xming 和 Xmanager 是收费软件。
+Xming 在 2007 年发布了最后一个免费版本（6.9.0.31）。
+VcXsrv 的使用方式和界面与 Xming 极为相近。
+推荐使用 VcXsrv，本文以此软件为例进行介绍。
+
+1.  下载 `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`__\ ，默认安装即可
+
+2.  运行 XLaunch，除了在 ``Extra settings`` 界面勾选 ``Disable access control`` 之外，其它选项无需更改
+
+3.  在 Linux 的 ``~/.bashrc`` 文件中添加如下内容，然后执行 ``source ~/.bashrc`` ::
+
+        # Vcxsrv Display
+        # Windows 每次重启后 WSL2 namesever 的 IP 可能发生变化，如下设置保证能够连接到 XServer
+        export DISPLAY=$(awk '/nameserver / {print $2; exit}' /etc/resolv.conf 2>/dev/null):0
+        export LIBGL_ALWAYS_INDIRECT=1
```

Comment:
Could the command line be used to do this instead in WSL?

```
$ echo "export DISPLAY=$(awk '/nameserver / {print $2; exit}' /etc/resolv.conf 2>/dev/null):0" >> ~/.bashrc
$ echo "export LIBGL_ALWAYS_INDIRECT=1" >> ~/.bashrc
$ source ~/.bashrc
```

Reply:
I‘m afraid not, it will return a result to the `.bashrc` file like `export DISPLAY=172.17.179.65:0`

Reply:
Great